File: app/src/main/java/com/mgaetan89/showsrage/presenter/EpisodePresenter.java
package com.mgaetan89.showsrage.presenter;

import android.support.annotation.ColorRes;
import android.support.annotation.Nullable;
import android.text.format.DateUtils;

import com.mgaetan89.showsrage.helper.DateTimeHelper;
import com.mgaetan89.showsrage.model.Episode;

public class EpisodePresenter {
	@Nullable
	private Episode episode;

	public EpisodePresenter(@Nullable Episode episode) {
		this.episode = episode;
	}

	@Nullable
	public CharSequence getAirDate() {
		if (this.episode == null) {
			return null;
		}

		String airDate = this.episode.getAirDate();

		if (airDate == null || airDate.isEmpty()) {
			return null;
		}

		return DateTimeHelper.getRelativeDate(airDate, "yyyy-MM-dd", DateUtils.DAY_IN_MILLIS);
	}

	public String getQuality() {
		if (this.episode == null) {
			return "";
		}

		String quality = this.episode.getQuality();

		if ("N/A".equalsIgnoreCase(quality)) {
			return "";
		}

		return quality;
	}

	@ColorRes
	public int getStatusColor() {
		if (this.episode == null) {
			return android.R.color.transparent;
		}

		return this.episode.getStatusBackgroundColor();
	}
}

File: app/src/main/kotlin/com/mgaetan89/showsrage/presenter/EpisodePresenter.kt
package com.mgaetan89.showsrage.presenter

import android.support.annotation.ColorRes
import android.text.format.DateUtils
import com.mgaetan89.showsrage.extension.toRelativeDate
import com.mgaetan89.showsrage.model.Episode

class EpisodePresenter(val episode: Episode?) {
	fun getAirDate(): CharSequence? {
		val airDate = this._getEpisode()?.airDate ?: return null

		if (airDate.isEmpty()) {
			return null
		}

		return airDate.toRelativeDate("yyyy-MM-dd", DateUtils.DAY_IN_MILLIS)
	}

	fun getQuality(): String {
		val quality = this._getEpisode()?.quality ?: return ""

		return if ("N/A".equals(quality, true)) "" else quality
	}

	@ColorRes
	fun getStatusColor() = this._getEpisode()?.getStatusBackgroundColor() ?: android.R.color.transparent

	private fun _getEpisode() = if (this.episode?.isValid == true) this.episode else null
}
